### ISSUE TITLE
kademlia-1.1.0.0 test suite runs too long

### ISSUE THREAD

**Issue by peti** [froozen/kademlia#2]
Hi, we packaged `kademlia` on NixOS, but unfortunately our central builds fail with a timeout error during the test suite run. Is it normal for that test suite to run for 2+ hours? Please see http://hydra.nixos.org/build/25334361/log/raw for a complete build log.

**Comment by froozen** [comment]
The tests take 55 seconds on my laptop, so I can't realy imagine it taking 2+ hours, even on a busy hydra.
This probably means that something goes wrong during those tests, but I can't think of anything that could cause the tests not to finish.
There is a **lot** of stuff being sent between sockets during the tests, could that be a problem on a hydra?

**Comment by peti** [comment]
Does the test suite depend on a nameserver or an public Internet access? Because those options are unavailable to Hydra builds.

**Comment by froozen** [comment]
Nope, all the communication is between UDP sockets on localhost in the portrange of 1123-1142.
A failure to connect should result in a failure of the test, though, so that shouldn't be a problem.

**Comment by peti** [comment]
Well, in that case I'm not sure how to explain this phenomenon. The test ran on i686. Maybe that makes a difference?

**Comment by photm5** [comment]
I’m currently running the tests on my ThinkPad X40 (Which is i686). I started them about 20 minutes ago, and they are still running.

**EDIT**: I killed it after about 2 hours and 50 minutes as it started to eat my memory. (It was at 1G swap already)

**Comment by idcm** [comment]
The same happens for windows 10 (which is x86-64).
